pkg/lib/aws/autoscaling.go

```diff
@@ -59,3 +63,37 @@ func (c *Client) AutoscalingGroups(tags map[string]string) ([]*autoscaling.Group
 
 	return asgs, nil
 }
+
+// Return all activities for an ASG which do not have the "Successful" status, sorted from most to least recent
+// Also returns true if the most recent activity for the ASG is "Successful", or if there are no activities for the ASG
+func (c *Client) AutoscalingGroupUnsuccessfulActivities(asgName string) ([]*autoscaling.Activity, bool, error) {
```

Comment:
It looks like this function can return the most recent activity. The caller can inspect the most recent activity and determine if it is successful.